### ISSUE TITLE
`missing-docs` lint reports on `#[test]` macro

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust#130216]
I tried this code:

```rust
#![deny(missing_docs)]

//! Docs

#[test]
fn foo() {}
```

I expected to see this happen: no errors

Instead, this happened:

```console
> cargo check --all-targets
error: missing documentation for a constant
 --> src/lib.rs:6:1
  |
5 | #[test]
  | ------- in this procedural macro expansion
6 | fn foo() {}
  | ^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![deny(missing_docs)]
  |         ^^^^^^^^^^^^
  = note: this error originates in the attribute macro `test` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Meta
`rustc --version --verbose`:
```
rustc 1.83.0-nightly (0ee7cb5e3 2024-09-10)
binary: rustc
commit-hash: 0ee7cb5e3633502d9a90a85c3c367eccd59a0aba
commit-date: 2024-09-10
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0
```

Very likely to be https://github.com/rust-lang/rust/pull/130025 related

**Comment by Nemo157** [comment]
Looks like [`a1a8627` (#130025)](https://github.com/rust-lang/rust/pull/130025/commits/a1a8627dd789bd7444c78b7a5d360328709a54eb) allowed missing docs on the generated `main` function, but didn't test having any actual `#[test]`s.

cc @Urgau

**Comment by ogoffart** [comment]
Attempt fix in https://github.com/rust-lang/rust/pull/130219
